lib/log.h

```diff
@@ -77,10 +77,17 @@ extern void metal_default_log_handler(enum metal_log_level level,
  * @param	level	Log level.
  * @param	...	Format string and arguments.
  */
-#define metal_log(level, ...)						       \
-	((level <= _metal.common.log_level && _metal.common.log_handler) \
-	       ? (void)_metal.common.log_handler(level, __VA_ARGS__)	       \
-	       : (void)0)
+#define metal_log(level, fmt, args ...)					\
+do {									\
+	if (_metal.common.log_handler && level <= _metal.common.log_level) \
+		_metal.common.log_handler(level, "%s:%u " fmt "\n",	\
+					  __func__, __LINE__, ##args);	\
+} while (0)
+
+#define ML_ERR(fmt, args ...) metal_log(METAL_LOG_ERROR, fmt, ##args)
```

Comment:
Can we have macros like ml_err, ml_info, ml_debug, ml_warn.
These names are matching with kernel pr_* macros.
It will be easy for users to stay align with names that are matching what kernel has.